### PR TITLE
Change error message for Expression.ArrayLength to be more accurate.

### LINKED ISSUE #14406: Documentation and error message on ArrayLength is misleading
If one does:

```C#
Array.CreateInstance(typeof(int), new []{10}, new[]{-5})
```

Then the error message is "Argument must be single dimensional array type", as it was prior to #4164 blocking such expressions (previously it would be accepted but error on compilation), even though the array is indeed single-dimensional. It should be something like "Argument must be single-dimensional zero-based array type".

### PR DESCRIPTION
**Pull request by JonHanna**
Update XML documentation as well.

Fixes #14406.

### CODE REVIEW

File: src/System.Linq.Expressions/src/Resources/Strings.resx [hunk 1]
```diff
@@ -280,7 +280,7 @@
     <value>Argument for array index must be of type Int32</value>
   </data>
   <data name="ArgumentMustBeSingleDimensionalArrayType" xml:space="preserve">
-    <value>Argument must be single dimensional array type</value>
+    <value>Argument must be single-dimensional zero-based array type</value>
```
Comment (stephentoub) [inline src/System.Linq.Expressions/src/Resources/Strings.resx]: Nit: could you add a comma after dimensional?

Comment (JonHanna) [inline src/System.Linq.Expressions/src/Resources/Strings.resx]: Sure.

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop Windows_NT Debug Build and Test please (https://github.com/dotnet/corefx/issues/14409)

**Comment by stephentoub** [comment]
Thanks, @JonHanna.

**Pull request merged**